api/pkg/utils/encryptionutils.go

```diff
@@ -0,0 +1,78 @@
+package utils
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+)
+import log "github.com/sirupsen/logrus"
+
+func EncryptWithAES256RandomKey(data []byte, key []byte) (error, []byte) {
+	// use the key, get the cipher
+	cipherBlock, cipherErr := aes.NewCipher(key)
+	if cipherErr != nil {
+		log.Errorf("Encryption error, cipher not generated")
+		return cipherErr, nil
+	}
+	// use the cipher block to encrypt
+	// gcm or Galois/Counter Mode, is a mode of operation
+	// for symmetric key cryptographic block ciphers
+	// https://en.wikipedia.org/wiki/Galois/Counter_Mode
+	aesgcm, gcmErr := cipher.NewGCM(cipherBlock)
+	if gcmErr != nil {
+		log.Errorf("Encryption error, GCM not generated")
+		return gcmErr, nil
+	}
+	nonce := make([]byte, aesgcm.NonceSize())
+	_, nonceErr := io.ReadFull(rand.Reader, nonce)
+	if nonceErr != nil {
+		log.Errorf("Encryption error, GCM nonce not createdf")
+		return nonceErr, nil
+	}
+
+	// use the aes gcm to seal the data
+	encBytes := aesgcm.Seal(nonce, nonce, data, nil)
+	return nil, encBytes
+}
+
+func DecryptWithAES256(data []byte, key []byte) (error, []byte) {
+	// use the key, get the cipher
+	cipherBlock, cipherErr := aes.NewCipher(key)
+	if cipherErr != nil {
+		log.Errorf("Decryption error, cipher not generated")
+		return cipherErr, nil
+	}
+	// use the cipher block to decrypt
+	// gcm or Galois/Counter Mode, is a mode of operation
+	// for symmetric key cryptographic block ciphers
+	// https://en.wikipedia.org/wiki/Galois/Counter_Mode
+	aesgcm, gcmErr := cipher.NewGCM(cipherBlock)
+	if gcmErr != nil {
+		log.Errorf("Decryption error, GCM not generated")
+		return gcmErr, nil
+	}
+	nonceSize := aesgcm.NonceSize()
+	nonce, ciphertext := data[:nonceSize], data[nonceSize:]
```

Comment:
do we need to check whether the length of data is more than the nonce size here?

Reply:
not needed, the nonce is pre-pended to the encrypted data, so the data[] will be nonce-size + actual-data-size

Reply:
ok, if user uploads  some malformed data whose size is smaller than nonce-size, can we block this beforehand and make the program not crash?

Reply:
sure, raised issue to track this, will test and fix #773